### PR TITLE
chore: initial draft for stream configs with hints

### PR DESCRIPTION
**Pull request by nitroin**
This is a initial draft with some ideas to implements query hints, addressing related #320 issue.

I need to understand how to correctly test this code but I will wait for feedback before proceed further.

### CODE REVIEW

File: streamconfiguration/src/main/java/com/speedment/jpastreamer/streamconfiguration/StreamConfiguration.java [hunk 1]
```diff
@@ -130,6 +132,10 @@ default StreamConfiguration<T> joining(final Field<T> field) {
      */
     StreamConfiguration<T> selecting(final Projection<T> projection);
 
+    Map<String, Object> hints();
```
Comment (julgus) [inline streamconfiguration/src/main/java/com/speedment/jpastreamer/streamconfiguration/StreamConfiguration.java:135]: Please add JavaDoc

File: streamconfiguration/src/main/java/com/speedment/jpastreamer/streamconfiguration/StreamConfiguration.java [hunk 2]
```diff
@@ -130,6 +132,10 @@ default StreamConfiguration<T> joining(final Field<T> field) {
...
+
+    StreamConfiguration<T> withHint(final String hintName, final Object value);
```
Comment (julgus) [inline streamconfiguration/src/main/java/com/speedment/jpastreamer/streamconfiguration/StreamConfiguration.java:137]: Please add JavaDoc

**Review by julgus — CHANGES_REQUESTED** [review]
Sorry for the delayed response. I think this looks great! Please just write JavaDocs for the added `StreamConfiguration` methods and I'll merge this in. 

These things are always tricky to test. For now, we rely heavily on the jpastreamer-demo repo for manual integration tests before release.

**Comment by nitroin** [comment]
Added initial javadoc, LGTY?

**Comment by nitroin** [comment]
I'll open a separate PR for `jpastreamer-demo` to cross validate this change.

**Comment by julgus** [comment]
> I'll open a separate PR for `jpastreamer-demo` to cross validate this change.

Sounds great!

**Comment by julgus** [comment]
> Added initial javadoc, LGTY?

Looking good! Did you see my note about signing the Contributor Licence Agreement? See instructions here: [https://github.com/speedment/jpa-streamer/blob/master/CONTRIBUTING.md](https://github.com/speedment/jpa-streamer/blob/master/CONTRIBUTING.md). The signed agreement should be added along with your PR. 

Thanks again!

**Comment by nitroin** [comment]
This should be ready to go!

_It's works on my machine!_

File: cla/nitroin [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+either might otherwise have in any such action or proceeding in any of such courts based
+upon any alleged lack of personal jurisdiction, improper venue, forum non conveniens or
+any similar claim or defense. This Agreement sets forth the entire understanding and
+agreement between the parties, and supersedes any previous communications,
+representations or agreements, whether oral or written, regarding the subject matter
+herein.
+
+
+
+Please sign: Manuel Serra_______________ Date: 19/05/2022______
```
Comment (julgus) [inline cla/nitroin:138]: Please update to 2023

**Review by julgus — APPROVED** [review]

**Pull request merged**